services/src/main/java/io/grpc/services/CallMetricRecorder.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2019 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.services;
+
+import io.grpc.Context;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicReference;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Utility to record call metrics for load-balancing. One instance per call.
+ */
+@ThreadSafe
```

Comment:
`@since`
`@ExperimentalApi`

Reply:
Or package private?

Reply:
Done with `@ExperimentalApi`. We currently do not have a version number when this API can be used for open-source users.

Reply:
We add `@since` to public API regardless if it can be useful for open-source users.